src/python/grpcio_tests/tests/http2/_negative_http2_client.py

```diff
@@ -0,0 +1,158 @@
+  elif len(response.payload.body) != expected_length:
+    raise ValueError(
+      'expected payload body size %d, got %d' %
+          (expected_length, len(response.payload.body)))
+
+def _expect_status_code(call, expected_code):
+  if call.code() != expected_code:
+    raise ValueError(
+      'expected code %s, got %s' % (expected_code, call.code()))
+
+def _expect_status_details(call, expected_details):
+  if call.details() != expected_details:
+    raise ValueError(
+      'expected message %s, got %s' % (expected_details, call.details()))
+
+def _validate_status_code_and_details(call, expected_code, expected_details):
+  _expect_status_code(call, expected_code)
+  _expect_status_details(call, expected_details)
+
+# common requests
+_SMALL_SIZE = 1024
+_LARGE_SIZE = 314159
+
+_SMALL_SIMPLE_REQUEST = messages_pb2.SimpleRequest(
+    response_type=messages_pb2.COMPRESSABLE,
+    response_size=_SMALL_SIZE,
+    payload=messages_pb2.Payload(body=b'\x00' * _SMALL_SIZE))
+
+_STREAMING_OUTPUT_REQUEST = messages_pb2.StreamingOutputCallRequest(
+    response_type=messages_pb2.COMPRESSABLE,
+    response_parameters=(
+        messages_pb2.ResponseParameters(size=_SMALL_SIZE),))
+
+def _goaway(stub):
+  first_response = stub.UnaryCall(_SMALL_SIMPLE_REQUEST)
+  _validate_payload_type_and_length(first_response, 
+      messages_pb2.COMPRESSABLE, _SMALL_SIZE)
+  second_response = stub.UnaryCall(_SMALL_SIMPLE_REQUEST)
+  _validate_payload_type_and_length(second_response, 
+      messages_pb2.COMPRESSABLE, _SMALL_SIZE)
+
+def _rst_after_header(stub):
+  resp_future = stub.UnaryCall.future(_SMALL_SIMPLE_REQUEST)
+  _validate_status_code_and_details(resp_future, grpc.StatusCode.UNKNOWN, "")
+
+def _rst_during_data(stub):
+  resp_future = stub.UnaryCall.future(_SMALL_SIMPLE_REQUEST)
+  _validate_status_code_and_details(resp_future, grpc.StatusCode.UNKNOWN, "")
+
+def _rst_after_data(stub):
+  resp_future = stub.UnaryCall.future(_SMALL_SIMPLE_REQUEST)
+  _validate_payload_type_and_length(next(resp_future),
+      messages_pb2.COMPRESSABLE, _SMALL_SIZE)
+  _validate_status_code_and_details(resp_future, grpc.StatusCode.UNKNOWN, "")
+
+def _ping(stub):
+  response = stub.UnaryCall(_SMALL_SIMPLE_REQUEST)
+  _validate_payload_type_and_length(response, 
+      messages_pb2.COMPRESSABLE, _SMALL_SIZE)
+
+def _max_streams(stub):
+  # send one req to ensure server sets MAX_STREAMS
+  response = stub.UnaryCall(_SMALL_SIMPLE_REQUEST)
+  _validate_payload_type_and_length(response, 
+      messages_pb2.COMPRESSABLE, _SMALL_SIZE)
+
+  # give the streams a workout
+  futures = []
+  for _ in range(15):
+    futures.append(stub.UnaryCall.future(_SMALL_SIMPLE_REQUEST))
+  for future in futures:
+    _validate_payload_type_and_length(future.result(),
+        messages_pb2.COMPRESSABLE, _SMALL_SIZE)
+
+def _run_test_case(test_case, stub):
+  if test_case == 'goaway':
+    _goaway(stub)
+  elif test_case == 'rst_after_header':
```

Comment:
Java client uses all upper case letters for the test cases (https://github.com/grpc/grpc-java/pull/2513/files). Let's be consistent.